packages/plugins/src/hardhat/deployments.ts

```diff
@@ -336,23 +335,32 @@ export const proposeChugSplashBundle = async (
   configUri: string,
   remoteExecution: boolean,
   ipfsUrl: string,
-  isUpgrade: boolean,
-  configPath: string
+  configPath: string,
+  spinner: ora.Ora
 ) => {
+  // Determine if the deployment is an upgrade
+  const isUpgrade = await checkIsUpgrade(hre.ethers.provider, parsedConfig)
   if (isUpgrade) {
+    spinner.succeed('Upgrade detected')
+    // Check if upgrade is valid
     await checkValidUpgrade(
       hre.ethers.provider,
       parsedConfig,
       configPath,
       hre.network.name
     )
-  } else {
-    await checkValidDeployment(
-      hre.ethers.provider,
-      parsedConfig,
-      configPath,
-      hre.network.name
-    )
+
+    // Confirm upgrade with user
+    const confirm = await yesno({
+      question: `Prior deployments detected for project ${parsedConfig.options.projectName}, would you like to perform an upgrade? (y/n)`,
+    })
+
+    if (!confirm) {
+      throw new Error(
+        `The project name ${parsedConfig.options.projectName} was already used in a previous deployment. To perform a fresh deployment
+of a new project, you must change the project name to something other than ${parsedConfig.options.projectName}.`
+      )
```

Comment:
Could you add something to the beginning of the error message to affirm that the user denied the upgrade? E.g: `User denied upgrade. The project name ${projectName} was already used...`